### PR TITLE
Infra: Improve Bug report template

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

**Review by amogh-jahagirdar — APPROVED** [review]
Love the addition, it'll help disambiguate in the community when people intend to put out a fix, and are simply notifying the existence of an issue vs reporting an issue.

**Review by RussellSpitzer — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/iceberg_bug_report.yml [hunk 1]
```diff
@@ -76,3 +76,11 @@ body:
         You can include files by dragging and dropping them here.
     validations:
       required: true
+  - type: checkboxes
+    attributes:
+      label: Willingness to contribute
+      description: The Iceberg community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the Iceberg codebase?
```
Comment (jbonofre) [inline .github/ISSUE_TEMPLATE/iceberg_bug_report.yml:82]: nit: I would rather use Apache Iceberg instead of Iceberg.

**Review by jbonofre — CHANGES_REQUESTED** [review]

**Review by ajantha-bhat — APPROVED** [review]

**Review by dramaticlly — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/iceberg_bug_report.yml (hunk 1 above)
Comment (nastra) [inline .github/ISSUE_TEMPLATE/iceberg_bug_report.yml:82]:
```suggestion
      description: The Apache Iceberg community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the Apache Iceberg codebase?
```

Comment (nastra) [reply .github/ISSUE_TEMPLATE/iceberg_bug_report.yml]: updated

**Review by jbonofre — APPROVED** [review]

**Pull request merged**